README.md

```diff
@@ -51,3 +51,47 @@ this.link = ApolloLink.from([
     new HttpLink({ uri: URI_TO_YOUR_GRAPHQL_SERVER }),
 ]);
 ```
+
+### Deduplicate
+
+It's possible to deduplicate operations while the queue is closed.
+This for example allows to only save the newest changes without the steps between.
+
+```js
+import { ApolloLink } from 'apollo-link';
+import { BatchHttpLink } from 'apollo-link-batch-http';
+import QueueLink from 'apollo-link-queue';
+
+const offlineLink = new QueueLink({
+    /**
+     * Decides which entry to keep in the queue in case of duplicate entries.
+     * Possible values:
+     *  - last: removes existing duplicate
+     *  - first: ignores new duplicate
+     *  - all: doesn't deduplicate
+     *
+     * Defaults to 'all'.
+     */
+    keepPolicy: 'last',
+
+    /**
+     * Specifies which entries are considered duplicates
+     *
+     * Defaults to comparing operation operationA.toKey() === operationB.toKey()
+     * https://www.apollographql.com/docs/link/overview.html
+     */
+    isDuplicate: (a, b) => (a.operationName === 'save' &&
```

Comment:
As a consumer I would find it useful to specify the `keepPolicy` per-query using a `context` option.

Also useful would be specifying a `dedupeKey`, defaulting to `${operationName}:${variables.id}`. `dedupeKey` would be used to calculate which operations `keepPolicy` should be applied to, sort of like a named queue. I hope that makes sense!

Reply:
I don't think it is a good idea to default to `${operationName}:${variables.id}` as this wouldn't fit most usecases.
* There isn't always an variable id.
* Maybe it's called differently.
* We for example use input types which would break this default es well.

The default should always work and if you have a more specific need you can overwrite the default.

Reply:
You're right, by default the `dedupeKey` should be undefined and operations shouldn't be deduped in the queue. What I meant to say is the recommended `dedupeKey` would be `${operationName}:${variables.id}`.

The use case I have in mind is editing multiple text documents and saving the edits while offline. I want to only save the final revision, which fits with `keepPolicy: 'last'`. However I want to keep the last operation for all different saved documents, so you would calculate which to keep with a `dedupeKey: 'saveDocument:${id}'`. Then when the queue opens it has exactly one `saveDocument` operation per unique document saved.

Reply:
👍 Yeah I'm doing exactly the same thing in our project. I'm just arguing that this shouldn't be the default.